src/configure/utilities/InputControlProvider.ts

```diff
@@ -0,0 +1,180 @@
+import { MustacheHelper } from "../helper/mustacheHelper";
+import { ExtendedInputDescriptor, ExtendedPipelineTemplate, InputDataType, InputMode } from "../model/Contracts";
+import { AzureSession, ControlType, IPredicate, StringMap } from '../model/models';
+import { DataSourceExpression } from "./DataSourceExpression";
+import { InputUxDescriptor } from "./InputUxDescriptor";
+import { InputUxDescriptorUtility } from "./InputUxDescriptorUtility";
+import { VisibilityHelper } from "./VisibilityHelper";
+
+export class InputControlProvider {
+    private _pipelineTemplate: ExtendedPipelineTemplate;
+    private _inputUxDescriptors: Map<string,InputUxDescriptor>;
+    private _uxInputs: Map<string,ExtendedInputDescriptor>;
+
+    constructor(pipelineTemplateParameters: ExtendedPipelineTemplate, inputs: { [key: string]: any}){
+        this._pipelineTemplate = pipelineTemplateParameters;
+        this._inputUxDescriptors = new Map<string, InputUxDescriptor>();
+        this._uxInputs = new Map<string, ExtendedInputDescriptor>();
+        this._createControls(inputs);
+    }
+
+    public async getAllInputUxDescriptors(azureSession: AzureSession,  ) {
+        let parameters: { [key: string]: any} = {};
+        for(let inputUxDescriptor of this._inputUxDescriptors.values()){
+            this._setInputUxVisibility(inputUxDescriptor);
+            this._setupDefaultInputValue(inputUxDescriptor);
+            await inputUxDescriptor.setInputUxDescriptorValue(azureSession);
+            parameters[inputUxDescriptor.getInputUxDescriptorId()] = inputUxDescriptor.getParameterValue();
+        }
+        return parameters;
+    }
+
+    private _createControls(inputValues: { [key: string]: any}) {
+        for( let input of this._pipelineTemplate.inputs){
+            this._uxInputs.set(input.id, input);
+            var inputUxDescriptor: InputUxDescriptor = null;
```

Comment:
you should change these names accordingly as per the type

Reply:
done

Reply:
Rethinking about this as almost all variables are named similarly and hence it will really be a very big change. You can choose to ignore this name change.